pkg/security/security_profile/tests/activity_tree_test.go

```diff
@@ -0,0 +1,635 @@
+			resultErr:             nil,
+		},
+
+		// try to insert a fileless root node
+		{
+			testName:              "try-insert-fileless-root-node",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/foo",
+			processPath:           "/bin/bar",
+			completeLineage:       true,
+			differentContainerID:  false,
+			fileLessParent:        true,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
+
+		// try to insert a fileless node
+		{
+			testName:              "try-insert-fileless-node",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/foo",
+			processPath:           "/bin/bar",
+			completeLineage:       true,
+			differentContainerID:  false,
+			fileLess:              true,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
+
+		// try to insert the granpa node inside the container
+		{
+			testName:              "try-insert-init-in-container",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/foo",
+			processPath:           "/bin/bar",
+			granpaInsideContainer: true,
+			completeLineage:       true,
+			differentContainerID:  false,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
+
+		// insert a runc node
+		{
+			testName:              "insert-runc-node",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/foo",
+			processPath:           "/bin/runc",
+			completeLineage:       true,
+			differentContainerID:  false,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
+
+		// try insert a runc node and parent node
+		{
+			testName:              "insert-runc-node-and-root",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/runc",
+			processPath:           "/bin/runc",
+			completeLineage:       true,
+			differentContainerID:  false,
+			resultNodeShouldBeNil: true,
+			resultNewProcessNode:  false,
+			resultErr:             activity_tree.ErrNotValidRootNode,
+		},
+
+		// try insert a runc node and parent node, will instead insert child node as root
+		{
+			testName:              "insert-runc-root",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/runc",
+			processPath:           "/bin/bar",
+			completeLineage:       true,
+			differentContainerID:  false,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
+
+		// insert fileless runc as root node // TODO: not sure if it's the expected result
+		{
+			testName:              "insert-runc-root",
+			resetActivityTree:     true,
+			parentProcessPath:     "/bin/runc",
+			processPath:           "/bin/bar",
+			completeLineage:       true,
+			differentContainerID:  false,
+			fileLessParent:        true,
+			resultNodeShouldBeNil: false,
+			resultNewProcessNode:  true,
+			resultErr:             nil,
+		},
```

Comment:
I'm still not sure about this behavior: having a fileless runc added as root node without errors

Reply:
Logic updated after discussion